notion-server/src/ethereum.rs

```diff
@@ -1,4 +1,77 @@
+use base64;
 use std::collections::HashMap;
+use std::convert::TryFrom;
+use snafu::{Snafu, ResultExt, Backtrace};
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    Decode {
+        backtrace: Backtrace,
+        source: base64::DecodeError,
+    },
+    OutOfBounds {
+        message: String,
+    }
+}
+
+pub mod interface_args {
+    #[derive(Debug, Default)]
+    pub struct GeneralSimulationState {
```

Comment:
Unless there is a need to differentiate `GeneralSimulationState` and `SimulationState`, I think I'd prefer the latter.

Reply:
Yeah I don't feel strongly here.  My thinking was that `SimulationState` could be interpreted to mean "everything the simulation state contains" vs. `GeneralSimulationState` felt more likely to be interpreted as "an broad overview of what the simulation state looks like".  Since this struct is meant to represent the latter, I chose the latter name.  Definitely open to discussing this or shortening it, though.